Realm/RLMRealm.mm

```diff
@@ -448,9 +448,19 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration
     }
 
     if (configuration.seedFilePath && ![RLMRealm fileExistsForConfiguration:configuration]) {
-        [[NSFileManager defaultManager] copyItemAtURL:configuration.seedFilePath
-                                                toURL:[configuration fileURL]
-                                                error:error];
+        bool didCopySeed = false;
+        bool didLock = DB::call_with_lock(configuration.seedFilePath.path.UTF8String,
+                                          [&configuration, &error, &didCopySeed](auto const&) {
+            didCopySeed = [[NSFileManager defaultManager] copyItemAtURL:configuration.seedFilePath
+                                                                  toURL:configuration.fileURL
+                                                                  error:error];
+        });
+        if (!didLock) {
+            @throw RLMException(@"Cannot copy seed Realm at path '%@' while it is already open.", configuration.seedFilePath);
+        }
+        if (!didCopySeed) {
+            @throw RLMException(@"Failed to copy seed Realm at path '%@' to destination path '%@'.", configuration.seedFilePath, configuration.fileURL);
```

Comment:
This is a runtime error and not a precondition, so it should set the `error` out parameter if that's non-nil and only throw an exception if it's nil (i.e. RLMSetErrorOrThrow). Either way the error message should include the error from `copyItemAtURL:toURL:` rather than discarding the underlying error.